fhir-persistence-schema/docs/DB2MultiTenancy.md

```diff
@@ -1,16 +1,225 @@
 
-## Multi-Tenancy Schema Design
+This document assumes familiarity with the general [IBM FHIR Server schema design](SchemaMigration.md).
+
+# Multi-Tenancy Concepts and Design
+
+The IBM FHIR Server design supports three levels of multi-tenancy:
+
+1. Separate database per tenant
+2. Separate data schema per tenant
+3. (Db2-only) multiple tenants per schema, partitioned with row-based access control and per-tenant tablespaces.
+
+Of course, an entire deployment (application servers and database) also could be dedicated to a single tenant, which is really just a variant of option 1 above, although clearly such a configuration would not be considered multi-tenant.
+
+## Tenant Data Administration
+
+The IBM FHIR Server separates schema administration data from user data. All administration data is held in tables within the FHIRADMIN schema, and these tables are not directly accessible by the IBM FHIR Server database user.
+
+The FHIRADMIN schema is used to manage schema version migrations as well as tenant configuration for any tenants resident in the given database. The FHIRADMIN schema does not manage tenants in other databases.
+
+See the general [IBM FHIR Server schema design](SchemaMigration.md) documentation for a more detailed description of how FHIRADMIN tables are used to manage schema versions.
+
+
+## Multi-tenancy using Db2 Table Partitions and Per Tenant Tablespaces
+
+The IBM FHIR Server provides a multi-tenancy feature based on Db2 row-based access control and table partitioning features.
+
+Each new tenant is allocated a unique tenant id value which is of database type `INT`. All tables in the IBM FHIR Server data schema (typically `FHIRDATA`) are created as range partitioned and configured with row-based access control. 
+
+As an example, consider a table of patients created using the following DDL (illustrative only, this is not the actual IBM FHIR Server schema):
+
+```
+CREATE TABLE ptng.patients (
+    mt_id               INT     NOT NULL,
+    patient_name    VARCHAR(18) NOT NULL,
+    age                 INT     NOT NULL,
+    doctor_name     VARCHAR(18),
+    ssn                CHAR(11),
+    CONSTRAINT pk_patient PRIMARY KEY (mt_id, patient_name)
+)
+PARTITION BY RANGE (mt_id) (STARTING 0 INCLUSIVE ENDING 0 INCLUSIVE);
+```
+
+Each tenant is allocated a unique integer value for mt_id (multi-tenant id), with a range partition used to define which table partition is to be used for a particular tenant's mt_id. The mapping of tenant name to mt_id value is maintained in the FHIR_ADMIN.TENANTS table.
+
+The above example assumes that patient_name will be unique within the context of a single tenant, which obviously does not apply to the actual IBM FHIR Server schema. The primary key must include the partition range key `mt_id`. In addition, any secondary indexes MUST also include mt_id to ensure they are always local to the partition. For example:
+
+```
+CREATE UNIQUE INDEX ptng.unq_ssn ON ptng.patients(mt_id, ssn);
```

Comment:
```suggestion
CREATE UNIQUE INDEX fhirdata.unq_ssn ON fhirdata.patients(mt_id, ssn);
```

Reply:
ended up changing the schema name to "example" to make it clearer.